Source/NSTextView.m

```diff
@@ -4982,6 +4982,43 @@ the range of characters (if any) to be replaced by the new data.  */
 	  RELEASE(attachment);
 	  return YES;
 	}
+      if ([type isEqualToString: NSFilenamesPboardType])
+	{
+          NSArray *list = [pboard propertyListForType: NSFilenamesPboardType];
+          NSMutableAttributedString *as = [[NSMutableAttributedString alloc] init]; 
+
+	  id<NSFastEnumeration> enumerator = list;
+	  FOR_IN (NSString*, filename, enumerator)
+	   {
+	      NSFileWrapper *fw = [[NSFileWrapper alloc] initWithPath:filename];
+	      if (fw) 
+	        {
+	          NSTextAttachment *attachment = [[NSTextAttachment alloc] 
+					         initWithFileWrapper: fw];
+	          NSAttributedString *asat =
+	            [NSAttributedString attributedStringWithAttachment: attachment];
+
+	          RELEASE(fw);
+	          RELEASE(attachment);
+
+	          [as appendAttributedString: asat];
+	        }
+	   }
+	  END_FOR_IN(enumerator)
+
+	  if (as && changeRange.location != NSNotFound &&
+	      [self shouldChangeTextInRange: changeRange
+		replacementString: [as string]])
+	    {
+	      [self replaceCharactersInRange: changeRange
+		withAttributedString: as];
+	      [self didChangeText];
+	      changeRange.length = [as length];
+	      [self setSelectedRange: NSMakeRange(NSMaxRange(changeRange),0)];
+	      RELEASE(as);
```

Comment:
You need to release "as" even if the other conditions are not met.

Reply:
done